src/SectionPage.tsx

```diff
@@ -0,0 +1,35 @@
+import * as types from './types';
+import {Files} from './Files';
+import {ChordProgression} from './ChordProgression';
+import {Comments} from './Comments';
+import {CreateComment} from './CreateComment';
+import {SectionTitle} from './SectionTitle';
+import {useGlobalStore} from './hooks/useGlobalStore';
+
+type SectionPageProps = {
+    projectId: string;
+    sectionId: string;
+}
+
+const SectionPage: React.FC<SectionPageProps> = ({projectId, sectionId}) => {
+    const globalStore = useGlobalStore();
+    const section = globalStore.getSection(sectionId);
+    const files = globalStore.getFilesForSection(sectionId);
+
+    const sectionPointer: types.EntityPointer = {
+        entityId: sectionId,
+        entityType: types.EntityType.SECTION,
+    };
+
+    return (
+        <div className="root">
+            <SectionTitle sectionId={sectionId} />
+            <ChordProgression chordProgression={section.chordProgression} />
+            <Files files={files} />
+            <Comments entityPointer={sectionPointer} />
+            <CreateComment entityPointer={sectionPointer} />
+        </div>
```

Comment:
A better design of how props would be passed in here may be like this:

```ts
    const globalStore = useGlobalStore();
    const actions = useActions();
    
    const section = globalStore.getSection(sectionId);
    const files = globalStore.getFilesForSection(sectionId);
    const comments = globalStore.getCommentsForSection(sectionId);

    const sectionPointer: types.EntityPointer = {
        entityId: sectionId,
        entityType: types.EntityType.SECTION,
    };
    
    const updateTitle = (newTitle: string) => {
        actions.updateSection(...);
    };
    
    const uploadFileToSection = (fileContent: File) => {
        actions.uploadFileToSection(sectionId, fileContent);
    };
    
    const addCommentToSection = (message: string) => {
        actions.addCommentToSection(sectionId, message);
    };
    
    return (
        <div className="root">
            <SectionTitle
                title={section.title}
                onEdit={updateTitle}
            />
            <ChordProgression
                chordProgression={section.chordProgression}
            />
            <Files
                files={files}
                uploadFile={uploadFileToSection}
            />
            <Comments
                comments={comments}
            />
            <CreateComment
                addComment={addCommentToSection}
            />
        </div>
```

Then the child components can be as "dumb" as possible https://devdojo.com/krissanawat101/minimal-explanation-of-dumb-and-smart-components-in-react

Reply:
The frontend is currently quite non-elastic, and will definitely need to be refactored in a lot of ways to make it more robust for different viewports, as well as supporting the functionality we're looking to implement. 

I think having a centralized way of managing all of these will go a long. Maybe we just make issues for each little issue like this, and then I can go through and clean them all up. What do you think is the best way to systematically tackle all of these?

Reply:
Yeah I think itemizing things is the way to go. I'll spend some time with that soon